src/qt3utils/applications/controllers/princeton_spectrometer.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import tkinter as tk
+import logging
+from typing import Tuple
+
+import qt3utils.datagenerators.princeton as princeton
+
+
+class QT3ScanPrincetonSpectrometerController:
+    """
+    Implements qt3utils.applications.qt3scan.interface.QT3ScanSpectrometerDAQControllerInterface
+    """
+    def __init__(self, logger_level: int):
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logger_level)
+
+        self.spectrometer = princeton.SpectrometerDataAcquisition()
+        self.last_config_dict = {}
+
+        self.wave_start = None
+        self.wave_end = None
+        self.last_measured_spectrum = None
+        self.last_wavelength_array = None
+
+    @property
+    def clock_rate(self) -> float:
+        try:
+            _t = self.spectrometer.exposure_time / 1000.0  # Converting from milliseconds to seconds.
+        except Exception as e:
+            self.logger.error(e)
+            _t = 2  # TODO: better default behavior. Should this be -1? 1? or should Spectrometer be changed.
```

Comment:
maybe np.nan is better?

Reply:
Not sure-- @cmordi we could revisit this and find out if this try block is still needed or if we fixed the problem. The problem relates to how the isinstance function of python works at this line: https://github.com/qt3uw/qt3-utils/blob/a99e4ef859cbbb69632351932afcda70b289d414/src/qt3utils/applications/qt3scan/main.py#L482.  The isinstance checks to see if a class implements all of the methods in a Protocol and I *think* for properties, it actually runs the method due to some python import reason. But we could try to remove the try/except block and see if it works. If it doesn't work without the try block, then yeah maybe np.nan works...